src/client/datascience/jupyter/kernels/cellExecutionJob.ts

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { IDisposable } from 'monaco-editor';
+import { NotebookCell, NotebookCellRunState, NotebookEditor } from '../../../../../types/vscode-proposed';
+import { disposeAllDisposables } from '../../../common/helpers';
+import { traceInfo } from '../../../common/logger';
+import { createDeferred } from '../../../common/utils/async';
+import { noop } from '../../../common/utils/misc';
+import { traceCellMessage } from '../../notebook/helpers/helpers';
+import { INotebook } from '../../types';
+import { CellExecution, CellExecutionFactory } from './cellExecution';
+// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
+const vscodeNotebookEnums = require('vscode') as typeof import('vscode-proposed');
+
+/**
+ * A job responsible for execution of cells.
+ * If this has not completed execution of the cells queued, we can continue to add more cells to this job.
+ * All cells queued using `runCells` are added to the queue and processed in order they were added/queued.
+ */
+export class CellExecutionJob {
```

Comment:
No idea why i kept calling it a stack, its not a stack, its a queue.
Renamed class to remove references to implementation detail.

Reply:
Maybe the name should be CellExecutionQueue then?